### ISSUE TITLE
Consolidate old&new stack for Modals

### ISSUE THREAD

**Issue by jardakotesovec** [pkp/pkp-lib#9870]
**Describe the issue**
As part of our old stack, we still have quite a lot of modals and dialogs being displayed via legacy stack, which is causing following challenges:

- combination of new and old stack modals does not work well together (issues with closing/controlling etc), which makes difficult scenarios with nested modals, where we need to modernise/add only part of that UI
- legacy modals don't have good accessibility handling (focus trapping inside modal) which is very confusing for screen reader users
- they look inconsistently mainly with our new side modals - therefore for example opening 'Assign participants' modal in new submission listing and workflow page would look differently (side modal vs central modal) even its same content, which is not great for UX

**Solution**
Render the legacy modals/dialogs via vue stack. Where tricky part is just correctly bridge the event flow between the modals and their creators (often grids).

Part of this work is also addressing downside of headless-ui library which requires to nest modals to be able correctly handle events and close correct one (might be improved in future upstream). As result for example if we need open dialog from SideModal, it needs to be dialog nested inside that SideModal, which is not practical in some cases (like displaying network error which is handled globally). 

Both shorcoming of headless-ui library and the legacy modal stack are addressed in `modalStore.js` for state management and `ModalsManager.vue` for rendering of the modals&dialogs.

From dev perspective its matter of interacting with simple API

```
   import {useModal} from '@/composables/useModal'

  const {openSideModal, openDialog} = useModal()


openSideModal(component, props);
openDialog(dialogProps)
```
Where the `component` either can be actual vue component instance or component name if that component is already registered. 

These examples will be in storybook.

**Outstanding**
We still have lots of vue3 `<modal>` components used, which are rendered centrally on the screen. And most likely we will want to migrate them to the SideModal and Dialog for consistent UX. But will keep this as separate issue as it will require some case by case considerations as there is couple of them which has very little content and will need to discuss with @Devika008 what shape&form would be fitting for these.

**Comment by jardakotesovec** [comment]
ojs: https://github.com/pkp/ojs/pull/4239
pkp-lib: https://github.com/pkp/pkp-lib/pull/9871
ui-library: https://github.com/pkp/ui-library/pull/344

Can I ask for review @asmecher @jonasraoni , thanks!


Outstanding for this issue (will finish tomorrow):
- add examples for sideModal to storybook using the openSideModal
- update OPS&OMP tests

**Comment by jardakotesovec** [comment]
Merged, closing.
